### ISSUE TITLE
cloudstack-setup-databases fails on complex database key

### ISSUE THREAD

**Issue by derhecht** [apache/cloudstack#4925]
##### ISSUE TYPE
 * Bug Report

##### COMPONENT NAME
~~~
Setup
~~~

##### CLOUDSTACK VERSION
~~~
4.15.0
~~~

##### CONFIGURATION
not relevant I think

##### OS / ENVIRONMENT
not relevant I think

##### SUMMARY
I run 
cloudstack-setup-databases cloud:<dbpassword>@localhost --deploy-as=root:<password> -e <encryption_type> -m <management_server_key> -k <database_key> -i <management_server_ip>
with e.g. the following (see database key)
cloudstack-setup-databases cloud:<dbpassword>@localhost --deploy-as=root:<password> -e <encryption_type> -m <management_server_key> -k "Some&Crazy&Password" -i <management_server_ip>
it splits the key and try to run it as commands (see screenshot)
![Bildschirmfoto 2021-04-15 um 15 27 36](https://user-images.githubusercontent.com/869755/114876987-27823c80-9dff-11eb-8546-024e1cec9ec0.jpg)

##### STEPS TO REPRODUCE

1. Follow install instructions on https://docs.cloudstack.apache.org/en/4.15.0.0/installguide/management-server/index.html
2. in step where you have to run cloudstack-setup-databases, do something like the following:

~~~
cloudstack-setup-databases cloud:"password"@localhost --deploy-as=root:"password" -m "password" -k "Some&Crazy&Password"
~~~

##### EXPECTED RESULTS
~~~
Script should setup the given key.
~~~

##### ACTUAL RESULTS
Error message when processing encryption.

~~~
see screenshot above
~~~

**Comment by rhtyd** [comment]
@derhecht which Linux distro and version is that? cc @davidjumani @shwstppr

**Comment by derhecht** [comment]
CentOS 7

**Comment by rhtyd** [comment]
@derhecht is this a containerised env, do you have `/bin/sh` as per the error in screenshot?

**Comment by derhecht** [comment]
yes, I have /bin/sh in my default Cent OS 7 installation

**Comment by derhecht** [comment]
I just changed the key and setup script works like a charm

**Comment by Spaceman1984** [comment]
> I just changed the key and setup script works like a charm

How do you mean @derhecht? Is this issue resolved then?

**Comment by derhecht** [comment]
no it is not resolved.
when i use a key within "&" (obviously) the script fails.
if not, script run successfully.

so, the content of the (secret) key affects the script called which should not.

**Comment by rhtyd** [comment]
@derhecht can you try the key string with ampersand with single quotes? The double quotes will cause the string to be evaluated, essentially it's a bash/shell thing than CloudStack issue.

**Comment by rhtyd** [comment]
cc @Spaceman1984 investigate and see if we can handle/fix something in the script (to handle keys with ampersand)

**Comment by derhecht** [comment]
> @derhecht can you try the key string with ampersand with single quotes? The double quotes will cause the string to be evaluated, essentially it's a bash/shell thing than CloudStack issue.

not atm as instance is running now - setup has finished - for the first try

**Comment by Spaceman1984** [comment]
> cc @Spaceman1984 investigate and see if we can handle/fix something in the script (to handle keys with ampersand)

I'm on it @rhtyd

**Comment by rhtyd** [comment]
Fixed in https://github.com/apache/cloudstack/pull/4976

**Comment by rhtyd** [comment]
@Spaceman1984 pl resubmit the PR against 4.15 branch

**Comment by rhtyd** [comment]
Fixed for 4.15 based on backport PR https://github.com/apache/cloudstack/pull/4995
